### ISSUE TITLE
Running ALTER TYPE ... ADD VALUE ... with prisma on cockroachDB causes prisma migrate dev/deploy to hang indefinitely

### ISSUE THREAD

**Issue by gautammmanek** [prisma/prisma#25254]
### Bug description

Generating a custom migration file with the following migration leads to the prisma migration engine hanging indefinitely (with cockroachDB):
`
-- AlterEnum
ALTER TYPE "QuestionType" ADD VALUE 'BINARY';
`

When running the migration in dev mode, debug mode produces the following:
`prisma:schemaEngine:stderr {"timestamp":"2024-09-21T11:10:32.091287Z","level":"INFO","fields":{"message":"Starting schema engine RPC server","git_hash":"b9a39a7ee606c28e3455d0fd60e78c3ba82b1a2b"},"target":"schema_engine"} +3ms`

Seems like the lack of a clear error message is similar to #24975 , where the actual error message faced by the engine is being obfuscated. Because of this, I can't verify the exact error produced when running the migration. 

### How to reproduce

Generate a migration file using npx prisma migrate dev --create-only after adding an additional value to an enum in the schema.prisma

Prisma generates the migration file with the following: ALTER TYPE "QuestionType" ADD VALUE 'BINARY'; (since the BINARY option was added to the QuestionType enum)

Running npx prisma migrate dev (or deploy) causes the migration engine to hang with the "Starting schema engine RPC server" error message

Inside the _prisma_migrations table in the db, the migration 20240919110944_add_yes_no_question containing the ALTER TYPE statement has a finished_at of NULL and an applied_steps_count of 0. The rolled_back_at and logs are both NULL:

### Expected behavior

ALTER TYPE "QuestionType" ADD VALUE 'BINARY' should add the BINARY value to the QuestionType enum without the migration failing  

### Prisma information

```prisma
// Add your schema.prisma

model Item {
  id                   String       @id @default(cuid())
  name                 String
  itemType         QuestionType
}

enum QuestionType {
  QUANTITY
  PHOTO
  BINARY // recently added to the enum, which resulted in the failed migration
}
```

### Environment & setup

- OS: macOS
- Database: CockroachDB, using cockroachdb/cockroach:latest (v24.2.0)
- Node.js version: v20.2.0

### Prisma Version

```
prisma                  : 5.13.0
@prisma/client          : 5.13.0
Computed binaryTarget   : darwin-arm64
Operating System        : darwin
Architecture            : arm64
Node.js                 : v20.2.0
Query Engine (Node-API) : libquery-engine b9a39a7ee606c28e3455d0fd60e78c3ba82b1a2b (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli b9a39a7ee606c28e3455d0fd60e78c3ba82b1a2b (at node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.13.0-23.b9a39a7ee606c28e3455d0fd60e78c3ba82b1a2b
Default Engines Hash    : b9a39a7ee606c28e3455d0fd60e78c3ba82b1a2b
Studio                  : 0.500.0
Preview Features        : fullTextSearch
```

**Comment by gautammmanek** [comment]
Very, very weirdly, it seems like baselining the database via collapsing all the existing migrations allows this new migration file to execute without issues. This is a non-solution of course, just a matter of interest

**Comment by gautammmanek** [comment]
Any updates to this issue? @SevInf

**Comment by UdittLamba** [comment]
This is still an issue for me.

**Comment by gautammmanek** [comment]
Yeah, this is still an issue, just ran into this again with prisma 5
